gulpfile.js

```diff
@@ -215,10 +217,30 @@ const test = gulp.series(testUnitTests, testWeb);
 const testInt = gulp.series(testDebugger);
 
 async function packageVsix() {
-    fs.emptyDirSync(packagedir);
-    return vsce.createVSIX({
+    const setPackageInfo = async function(setPreviewState) {
+        const standardHeader = '# Power Platform Extension';
```

Comment:
nit: Since each config has its own settings, it may be more scalable to create a json object with each option in a property.

```js
const pkgOptions = {
   name: 'powerplatform-vscode',
   displayName: 'Power Platform Tools',
   description: 'Tooling to create Power Platform solutions & packages, manage Power Platform environments and edit Power Apps Portals',
   readmeHeader: '# Power Platform Extension',
};
if(setPreviewState) {
   pkgOptions = {
      name: 'powerplatform-vscode-preview',
      displayName: 'Power Platform Tools [PREVIEW]',
      description: 'Unsupported extension for testing Power Platform Tools',
      readmeHeader: '# Power Platform Tools [PREVIEW]\n\n## This extension is used for internal testing against targets such as vscode.dev which require Marketplace published extensions, and is not supported.',
   };
}
```

Then, when applying the options, you can focus on encoding without complicating with conditionals:
```js
await npm(['pkg', 'set', `name=${pkgOptions.name}`]);
await npm(['pkg', 'set', `displayName="${pkgOptions.displayName}"`]);
await npm(['pkg', 'set', `description="${pkgOptions.description}"'}`]);

gulp.src('README.md')
   .pipe(replace(standardHeader, pkgOptions.readmeHeader)
   .pipe(gulp.dest('./'));
```

Reply:
Updated, but we couldn't be quite as clean as the example you suggested.   
The readme replacement logic needs to not only know what it should place in the file, but also what the other config's string is as a replacement target.